### ISSUE TITLE
llama-finetune is broken

### ISSUE THREAD

**Issue by yousefabdel1727** [ggml-org/llama.cpp#18482]
This result shows up on termux (Samsung M33 5G)

./build/bin/llama-finetune --model /storage/emulated/0/ysf/files/models/model-135m-f32.gguf  --file /storage/emulated/0/ysf/files/models/dataset.txt -c 128 -b 4 -ub 4 --epochs 1           main: force disabling memory mapping because it would result in-read-only pointers to the weights                                                                                               main: force changing k cache type to f32 due to a lack of f16 support for OUT_PROD              main: force changing v cache type to f32 due to a lack of f16 support for OUT_PROD              build: 7462 (6ce3d8579) with Clang 21.1.5 for Android aarch64                                   common_init_result: fitting params to device memory, for bugs during this step try to reproduce them with -fit off, or provide --verbose logs if the bug only occurs with -fit on               llama_params_fit_impl: no devices with dedicated memory found                                   llama_params_fit: successfully fit params to free device memory                                 llama_params_fit: fitting params to free memory took 0.14 seconds                               llama_model_loader: loaded meta data with 38 key-value pairs and 272 tensors from /storage/emulated/0/ysf/files/models/model-135m-f32.gguf (version GGUF V3 (latest))                           llama_model_loader: Dumping metadata keys/values. Note: KV overrides do not apply in this output.                                                                                               llama_model_loader: - kv   0:                       general.architecture str              = llama                                                                                               llama_model_loader: - kv   1:                               general.type str              = model                                                                                               llama_model_loader: - kv   2:                               general.name str              = SmolLM 135M                                                                                         llama_model_loader: - kv   3:                       general.organization str              = HuggingFaceTB                                                                                       llama_model_loader: - kv   4:                           general.finetune str              = instruct-add-basics                                                                                 llama_model_loader: - kv   5:                           general.basename str              = SmolLM                                                                                              llama_model_loader: - kv   6:                         general.size_label str              = 135Mllama_model_loader: - kv   7:                            general.license str              = apache-2.0                                                                                          llama_model_loader: - kv   8:                   general.base_model.count u32              = 1   llama_model_loader: - kv   9:                  general.base_model.0.name str              = SmolLM 135M                                                                                         llama_model_loader: - kv  10:          general.base_model.0.organization str              = HuggingFaceTB                                                                                       llama_model_loader: - kv  11:              general.base_model.0.repo_url str              = https://huggingface.co/HuggingFaceTB/...                                                            llama_model_loader: - kv  12:                               general.tags arr[str,7]       = ["alignment-handbook", "trl", "sft", ...                                                            llama_model_loader: - kv  13:                           general.datasets arr[str,5]       = ["HuggingFaceTB/Magpie-Pro-300K-Filte...                                                            llama_model_loader: - kv  14:                          llama.block_count u32              = 30  llama_model_loader: - kv  15:                       llama.context_length u32              = 2048llama_model_loader: - kv  16:                     llama.embedding_length u32              = 576 llama_model_loader: - kv  17:                  llama.feed_forward_length u32              = 1536llama_model_loader: - kv  18:                 llama.attention.head_count u32              = 9   llama_model_loader: - kv  19:              llama.attention.head_count_kv u32              = 3   llama_model_loader: - kv  20:                       llama.rope.freq_base f32              = 10000.000000                                                                                        llama_model_loader: - kv  21:     llama.attention.layer_norm_rms_epsilon f32              = 0.000010                                                                                            llama_model_loader: - kv  22:                           llama.vocab_size u32              = 49152                                                                                               llama_model_loader: - kv  23:                 llama.rope.dimension_count u32              = 64  llama_model_loader: - kv  24:            tokenizer.ggml.add_space_prefix bool             = false                                                                                               llama_model_loader: - kv  25:               tokenizer.ggml.add_bos_token bool             = false                                                                                               llama_model_loader: - kv  26:                       tokenizer.ggml.model str              = gpt2llama_model_loader: - kv  27:                         tokenizer.ggml.pre str              = smollm                                                                                              llama_model_loader: - kv  28:                      tokenizer.ggml.tokens arr[str,49152]   = ["<|endoftext|>", "<|im_start|>", "<|...                                                            llama_model_loader: - kv  29:                  tokenizer.ggml.token_type arr[i32,49152]   = [3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, ...                                                            llama_model_loader: - kv  30:                      tokenizer.ggml.merges arr[str,48900]   = ["Ġ t", "Ġ a", "i n", "h e", "Ġ Ġ...                                                                llama_model_loader: - kv  31:                tokenizer.ggml.bos_token_id u32              = 1   llama_model_loader: - kv  32:                tokenizer.ggml.eos_token_id u32              = 2   llama_model_loader: - kv  33:            tokenizer.ggml.unknown_token_id u32              = 0   llama_model_loader: - kv  34:            tokenizer.ggml.padding_token_id u32              = 2   llama_model_loader: - kv  35:                    tokenizer.chat_template str              = {% for message in messages %}{{'<|im_...                                                            llama_model_loader: - kv  36:               general.quantization_version u32              = 2   llama_model_loader: - kv  37:                          general.file_type u32              = 0   llama_model_loader: - type  f32:  272 tensors                                                   print_info: file format = GGUF V3 (latest)                                                      print_info: file type   = all F32                                                               print_info: file size   = 513.13 MiB (32.00 BPW)                                                load: printing all EOG tokens:                                                                  load:   - 0 ('<|endoftext|>')                                                                   load:   - 2 ('<|im_end|>')                                                                      load:   - 4 ('<reponame>')                                                                      load: special tokens cache size = 17                                                            load: token to piece cache size = 0.3170 MB                                                     print_info: arch             = llama                                                            print_info: vocab_only       = 0                                                                print_info: no_alloc         = 0                                                                print_info: n_ctx_train      = 2048                                                             print_info: n_embd           = 576                                                              print_info: n_embd_inp       = 576                                                              print_info: n_layer          = 30                                                               print_info: n_head           = 9                                                                print_info: n_head_kv        = 3                                                                print_info: n_rot            = 64                                                               print_info: n_swa            = 0                                                                print_info: is_swa_any       = 0                                                                print_info: n_embd_head_k    = 64                                                               print_info: n_embd_head_v    = 64                                                               print_info: n_gqa            = 3                                                                print_info: n_embd_k_gqa     = 192                                                              print_info: n_embd_v_gqa     = 192                                                              print_info: f_norm_eps       = 0.0e+00                                                          print_info: f_norm_rms_eps   = 1.0e-05                                                          print_info: f_clamp_kqv      = 0.0e+00                                                          print_info: f_max_alibi_bias = 0.0e+00                                                          print_info: f_logit_scale    = 0.0e+00                                                          print_info: f_attn_scale     = 0.0e+00                                                          print_info: n_ff             = 1536                                                             print_info: n_expert         = 0                                                                print_info: n_expert_used    = 0                                                                print_info: n_expert_groups  = 0                                                                print_info: n_group_used     = 0                                                                print_info: causal attn      = 1                                                                print_info: pooling type     = 0                                                                print_info: rope type        = 0                                                                print_info: rope scaling     = linear                                                           print_info: freq_base_train  = 10000.0                                                          print_info: freq_scale_train = 1                                                                print_info: n_ctx_orig_yarn  = 2048                                                             print_info: rope_yarn_log_mul= 0.0000                                                           print_info: rope_finetuned   = unknown                                                          print_info: model type       = 256M                                                             print_info: model params     = 134.52 M                                                         print_info: general.name     = SmolLM 135M                                                      print_info: vocab type       = BPE                                                              print_info: n_vocab          = 49152                                                            print_info: n_merges         = 48900                                                            print_info: BOS token        = 1 '<|im_start|>'                                                 print_info: EOS token        = 2 '<|im_end|>'                                                   print_info: EOT token        = 2 '<|im_end|>'                                                   print_info: UNK token        = 0 '<|endoftext|>'                                                print_info: PAD token        = 2 '<|im_end|>'                                                   print_info: LF token         = 198 'Ċ'                                                          print_info: FIM REP token    = 4 '<reponame>'                                                   print_info: EOG token        = 0 '<|endoftext|>'                                                print_info: EOG token        = 2 '<|im_end|>'                                                   print_info: EOG token        = 4 '<reponame>'                                                   print_info: max token length = 162                                                              load_tensors: loading model tensors, this can take a while... (mmap = false)                    load_tensors:          CPU model buffer size =   513.13 MiB                                     ................................................................................                common_init_result: added <|endoftext|> logit bias = -inf                                       common_init_result: added <|im_end|> logit bias = -inf                                          common_init_result: added <reponame> logit bias = -inf                                          llama_context: constructing llama_context                                                       llama_context: n_seq_max     = 1                                                                llama_context: n_ctx         = 256                                                              llama_context: n_ctx_seq     = 256                                                              llama_context: n_batch       = 4                                                                llama_context: n_ubatch      = 4                                                                llama_context: causal_attn   = 1                                                                llama_context: flash_attn    = auto                                                             llama_context: kv_unified    = false                                                            llama_context: freq_base     = 10000.0                                                          llama_context: freq_scale    = 1                                                                llama_context: n_ctx_seq (256) < n_ctx_train (2048) -- the full capacity of the model will not be utilized                                                                                      llama_context:        CPU  output buffer size =     0.19 MiB                                    llama_kv_cache:        CPU KV buffer size =    11.25 MiB                                        llama_kv_cache: size =   11.25 MiB (   256 cells,  30 layers,  1/1 seqs), K (f32):    5.62 MiB, V (f32):    5.62 MiB                                                                            llama_context: Flash Attention was auto, set to enabled                                         llama_context:        CPU compute buffer size =     0.79 MiB                                    llama_context: graph nodes  = 997                                                               llama_context: graph splits = 1                                                                 common_init_from_params: warming up the model with an empty run - please wait ... (--no-warmup to disable)                                                                                                                                                                                      system_info: n_threads = 8 (n_threads_batch = 8) / 8 | CPU : NEON = 1 | ARM_FMA = 1 | FP16_VA = 1 | DOTPROD = 1 | LLAMAFILE = 1 | OPENMP = 1 | REPACK = 1 |                                     -optimizer adamw -lr0 1e-05 -wd 0 -lr-min -1 -min-epochs -1 -epochs 1 -period 1 -val 0.05       /data/data/com.termux/files/home/llama.cpp/ggml/src/ggml.c:6871: GGML_ASSERT(!node->view_src || node->op == GGML_OP_CPY || node->op == GGML_OP_VIEW || node->op == GGML_OP_RESHAPE || node->op == GGML_OP_PERMUTE || node->op == GGML_OP_TRANSPOSE) failed                                      0: 0x735d18d4a4                                                                                 1: 0x735d18d464 ggml_print_backtrace                                                            2: 0x735d19e7d4 ggml_abort                                                                      3: 0x735d199c8c ggml_build_backward_expand                                                      4: 0x735d1a9594                                                                                 5: 0x735d1a9d24 ggml_opt_alloc                                                                  6: 0x735d2cb664 _ZN13llama_context14opt_epoch_iterEP16ggml_opt_datasetP15ggml_opt_resultRKNSt6__ndk16vectorIiNS4_9allocatorIiEEEESA_R11llama_batchPFvbP16ggml_opt_contextS1_S3_lllEblll         7: 0x735d2cbab8 _ZN13llama_context9opt_epochEP16ggml_opt_datasetP15ggml_opt_resultS3_lPFvbP16ggml_opt_contextS1_S3_lllES7_                                                                      8: 0x589be6eda8                                                                                 9: 0x735f0ba280 __libc_init                                                                     Aborted                    ./build/bin/llama-finetune --model /storage/emulated/0/ysf/files/models/model-135m-f32.gguf --file /storage/emulated/0/ysf/files/models/dataset.txt -c 128 -b 4 -ub 4 --epochs 1                                                                                     ┌─[ysf@android:~/llama.cpp]                                                                     └─$
